test/math/SafeMath.test.js

```diff
@@ -1,60 +1,91 @@
-import assertRevert from '../helpers/assertRevert';
-const assertJump = require('../helpers/assertJump');
-var SafeMathMock = artifacts.require('SafeMathMock');
+import assertJump from '../helpers/assertJump';
+const BigNumber = web3.BigNumber;
+const SafeMathMock = artifacts.require('SafeMathMock');
 
-contract('SafeMath', function (accounts) {
-  let safeMath;
+require('chai')
+  .use(require('chai-bignumber')(BigNumber))
+  .should();
+
+contract('SafeMath', () => {
+  const MAX_UINT = new BigNumber('115792089237316195423570985008687907853269984665640564039457584007913129639935');
 
   before(async function () {
-    safeMath = await SafeMathMock.new();
+    this.safeMath = await SafeMathMock.new();
   });
 
-  it('multiplies correctly', async function () {
-    let a = 5678;
-    let b = 1234;
-    await safeMath.multiply(a, b);
-    let result = await safeMath.result();
-    assert.equal(result, a * b);
-  });
+  describe('add', function () {
+    it('adds correctly', async function () {
+      const a = new BigNumber(5678);
+      const b = new BigNumber(1234);
 
-  it('adds correctly', async function () {
-    let a = 5678;
-    let b = 1234;
-    await safeMath.add(a, b);
-    let result = await safeMath.result();
+      const result = await this.safeMath.add(a, b);
+      result.should.be.bignumber.equal(a.plus(b));
+    });
 
-    assert.equal(result, a + b);
+    it('throws an error on addition overflow', async function () {
+      const a = MAX_UINT;
+      const b = new BigNumber(1);
```

Comment:
can we test the case of `+1` and `+2` here? I feel like an overflow to 0, which obviously included, might be a special case worth checking?